packages/web3torrent/src/clients/payment-channel-client.ts

```diff
@@ -176,6 +176,24 @@ export class PaymentChannelClient {
 
   // payer may use this method to make payments (if they have sufficient funds)
   async makePayment(channelId: string, amount: string) {
+    const channelReady = new Promise(resolve => {
+      if (
+        this.channelCache[channelId] &&
+        this.isAcceptanceOfMyPayment(this.channelCache[channelId])
+      ) {
+        resolve();
+      } else {
+        this.channelClient.onChannelUpdated(channelResult => {
```

Comment:
Should unsubscribe get called before promise resolution?

Reply:
Good catch, see a0072ba